### PR TITLE
Add io.github.nichokas.g_up

### PR DESCRIPTION
**Pull request by Nichokas**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **[Link][https://github.com/Nichokas/g-up]**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by Nichokas** [comment]
bot, build org.nichokas.gup

**Comment by Nichokas** [comment]
[g-up-x86_64.zip](https://github.com/user-attachments/files/16698696/g-up-x86_64.zip)
[g-up-aarch64.zip](https://github.com/user-attachments/files/16698697/g-up-aarch64.zip)

**Comment by Nichokas** [comment]
[flatpak-builds.zip](https://github.com/user-attachments/files/16698717/flatpak-builds.zip)

**Review by hfiguiere — CHANGES_REQUESTED** [review]
A flathub submission is NOT to host upstream code.

**Comment by hfiguiere** [comment]
> Please describe the application briefly.

Also while you acknowledged it, you didn't do it.

File: org.nichokas.gup.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+    "id" : "org.nichokas.gup",
```
Comment (hfiguiere) [inline org.nichokas.gup.json:2]: `nichokas.org` doesn't exist.

Yes you did acknowledge:

> The domain used for the application ID is [controlled by the application developers](https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository) and the [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

Comment (hfiguiere) [reply org.nichokas.gup.json:2]: and this is the build failure.

File: org.nichokas.gup.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "g-up",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/Nichokas/g-up"
```
Comment (hfiguiere) [inline org.nichokas.gup.json:41]: you always need it have a `commit` and/or `tag` here.

another reason of the build failure.

File: org.nichokas.gup.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "g-up",
+    "finish-args" : [
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
```
Comment (hfiguiere) [inline org.nichokas.gup.json:15]: this is not allowed and will not work.

File: org.nichokas.gup.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+    "id" : "org.nichokas.gup",
+    "runtime" : "org.freedesktop.Platform",
+    "runtime-version" : "23.08",
+    "sdk" : "org.freedesktop.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "g-up",
+    "finish-args" : [
```
Comment (hfiguiere) [inline org.nichokas.gup.json:10]: doesn't this need to display something?

**Pull request closed without merging**